### ISSUE TITLE
[FEATURE] Geojson input

### ISSUE THREAD

**Issue by msbarry** [onthegomap/planetiler#46]
Add support for newline-separated geojson input files, for example contour lines generated by gdal-contour.

**Comment by durkie** [comment]
do you envision that planetiler might pick up where tippecanoe left off? i regularly use tippecanoe with newline separated geojson, and it's a great piece of software, but it doesn't seem like it's getting much maintenance/development these days.

**Comment by msbarry** [comment]
Interesting, I doesn't realize the last commit to tippecanoe was almost a year ago. My guess is it's deprecated in favor of their new proprietary tiling service. 

I think there are some jobs people use tippecanoe for that planetiler could handle, but not sure about all of the density-based geometry filtering. Planetiler actually looks more like an alternative to the tiling service. 

The initial motivation was talking with someone trying to generate contour lines from DEM data. They used gdal contour to generate 1x1 degree contour tiles then they tried to use tippecanoe but it fell over trying to hold the planet in memory. Planetiler should be able to fill that gap since it does the same job but only holds a single input and output tile in memory at a time.

Supporting more input formats, and exposing more through config (#18) so you can do more without writing Java code would open up handling more use-cases from tippecanoe.

**Comment by flother** [comment]
What do you think about supporting [GeoPackage](https://www.geopackage.org/) as an input format too? It's essentially a standardised SQLite database format — a bit like MBTiles — and it's supposed to be a replacement for the venerable shapefile. I've seen it used quite a bit for government data.

**Comment by msbarry** [comment]
Makes sense! As long as you can read shapes with key/value tags from it then we should be able to use it as an input source.

**Comment by wipfli** [comment]
Can planetiler handle shapefiles (https://en.wikipedia.org/wiki/Shapefile)?

The Swiss government distributes shapefiles (https://data.geo.admin.ch/) for example for natural protected areas. In the past, I  converted shapefiles to GeoJSON and then to mbtiles with `tippecanoe`.

**Comment by msbarry** [comment]
> Can planetiler handle shapefiles (https://en.wikipedia.org/wiki/Shapefile)?

Yes, in fact planetiler currently can only handle shapefiles, natural earth sqlite file, and .osm.pbf files.  The openmaptiles profile uses shapfiles for [water polygons](https://osmdata.openstreetmap.de/data/water-polygons.html) and [lakelines](https://github.com/lukasmartinelli/osm-lakelines):

https://github.com/onthegomap/planetiler/blob/4ad4d952a6e13587d2b670d09e20d11b8d09bcf7/planetiler-basemap/src/main/java/com/onthegomap/planetiler/basemap/BasemapMain.java#L28-L50

**Comment by HarelM** [comment]
I'm looking into adding geojson support, I'll look at @erik's PR to better understand where I need to change the code.
As far as I can tell JTS only handles single geometry parsing here, so it can't be used to convert `FeatureCollection` and `Feature`, right? 
https://github.com/locationtech/jts/blob/master/modules/io/common/src/main/java/org/locationtech/jts/io/geojson/GeoJsonReader.java
I found the following libraries:
https://github.com/Urban-Research-Lab/jts-geojson
https://github.com/geotools/geotools
https://github.com/opendatalab-de/geojson-jackson

Not sure which is the preferred one, geotools seems like the one with the highest stars...
Let me know if you are familiar with a dependency I can use to start hacking this reader into planetiler...

**Comment by wipfli** [comment]
cc @bdon since you have been looking into a geojson parser for the clipping branch https://github.com/protomaps/basemaps/pull/351/files#diff-f325f761549c50b4e2b6f1cb1e77dd8232d99a9f5d4abedc934cabdc3260a6aa

**Comment by HarelM** [comment]
Since geotools are used for shapefile, it was an easy choice to use it for geojson as well.
I've opened a PR: #1147.
